scripts/segment_by_correlation.py

```diff
@@ -0,0 +1,60 @@
+import sys
+import os
+import csv
+from pathlib import Path
+
+import numpy as np
+import scipy.cluster.hierarchy as spc
+
+if len(sys.argv) < 3:
+    print(f"USAGE: python {os.path.basename(__file__)} <correlation.txt> <output.tsv>")
+    sys.exit(1)
+
+correlation_file = sys.argv[1]
+output_file      = sys.argv[2]
+
+protein_name = Path(correlation_file).stem
+
+correlation_matrix = np.loadtxt(correlation_file)
+
+# Code adapted from: https://stackoverflow.com/a/76042335
+pdist_uncondensed = 1.0 - correlation_matrix
+pdist_condensed = np.concatenate([row[i+1:] for i, row in enumerate(pdist_uncondensed)])
+linkage = spc.linkage(pdist_condensed, method='complete')
+clusters = spc.fcluster(linkage, 0.5 * pdist_condensed.max(), 'distance')
```

Comment:
This uses 1 - correlation as a distance metric. I just copied it off of SO, so I can't say whether it's the "right" clustering method for the job. I'm gonna need to do a lot more reading.